### PR TITLE
Docs: update and tidy up UGDG for peer review

### LINKED ISSUE #381: Docs: update and tidy up UGDG for peer review

### PR DESCRIPTION
**Pull request by tjtanjin**
As described in title.

Closes #381

### CODE REVIEW

**Review by JulietTeoh — APPROVED** [review]
LGTM!

File: docs/UserGuide.md [hunk 1]
```diff
@@ -310,40 +310,46 @@ A shorthand for `GET` requests can be done without specifying `-x` and `-u` (for
...
-- **JSON (JavaScript Object Notation):** A lightweight format for data storage. A more detailed explanation can be found [here](#84-json-format)
+
+| Term                                         | Description                                               |
+| -------------------------------------------- | --------------------------------------------------------- |
+| **API (Application Programming Interface)** | An interface for two systems to interact with each other  |
+| **Endpoint** | The point of entry in a communication channel for two systems to interact with each other |
+| **JSON (JavaScript Object Notation)** | A lightweight format for data storage. A more detailed explanation can be found [here](#84-json-format) |
+| **CURL (Client URL)** | A command-line tool used in the transfer of data via different network protocols. |
+
+<div style="page-break-after: always;"></div>
```
Comment (ong6) [inline docs/UserGuide.md:352]: nice addition.

File: docs/_sass/minima/_layout.scss [hunk 2]
```diff
@@ -166,7 +166,7 @@
 .post-content h1 {
   @include relative-font-size(2.625);
   letter-spacing: -1px;
-  line-height: 1.15;
+  line-height: 0;
```
Comment (ong6) [inline docs/_sass/minima/_layout.scss:169]: why did you change this to 0?

**Review by ong6 — APPROVED** [review]
LGTM, the farming has begun

File: docs/_sass/minima/_layout.scss (hunk 2 above)
Comment (tjtanjin) [reply docs/_sass/minima/_layout.scss:169]: Reduce the spacing between the title (guide) and the subtitle (version number) xD

Comment (ong6) [reply docs/_sass/minima/_layout.scss:169]: Ohh, ok nice
